### PR TITLE
chore(ui-react-native): Add env support to example app

### PR DESCRIPTION
**Pull request by ioanabrooks**
#### Description of changes

This change adds support for env variables to react-native example app leveraging `react-native-dotenv` library.

#### Issue #, if available

N/A

#### Description of how you validated changes

`yarn run react-native-example ios`

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/react-native/types/env.d.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+declare module '@env' {
+  export const Greeting: string;
```
Comment (calebpollman) [inline examples/react-native/types/env.d.ts:2]: Can we use SCREAMING_SNAKE_CASE here as these will be constants?

Comment (ioanabrooks) [reply examples/react-native/types/env.d.ts]: Yes!

**Comment by ioanabrooks** [comment]
Updated readme as per offline discussion with Caleb, thank you Caleb for your suggestions!!

File: examples/react-native/README.md [hunk 2]
```diff
@@ -99,6 +99,42 @@ To include an Amplify UI package as a dependency add it to the `dependencies` fi
...
+```
+
+To use your newly added env variable:
+
+```ts
+// *.tsx
+import { MY_ENV_VARIABLE } from '@env';
+```
+
+If the example app is not picking up changes to the values in _.env_ close Metro and reset the cache (see troubleshooting section).
```
Comment (calebpollman) [inline examples/react-native/README.md:136]:
```suggestion
> If the example app is not picking up changes to the values in _.env_ close Metro and reset the cache (see troubleshooting section).
```

**Review by calebpollman — APPROVED** [review]
LGTM 🚢 Thanks @ioanabrooks!

**Review by 0618 — APPROVED** [review]

**Pull request merged**